.github/workflows/lint.yml

```diff
@@ -6,13 +6,10 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: psf/black@stable
+      - uses: actions/checkout@v4
+      - uses: astral-sh/ruff-action@v3
```

Comment:
If we switch to using has instead of tags, e.g.

```
         - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683. # 4.2.2
```

We can than setup Dependabot to keep those up to date automatically.